pyomo/solvers/plugins/solvers/xpress_direct.py

```diff
@@ -894,41 +1005,60 @@ def _postsolve(self):
         # see if there is a solution available - this may not always
         # be the case, both in LP and MIP contexts.
         if self._save_results:
-            """
-            This code in this if statement is only needed for backwards compatibility. It is more efficient to set
-            _save_results to False and use load_vars, load_duals, etc.
-            """
+            # This code in this if statement is only needed for backwards
+            # compatibility. It is more efficient to set _save_results to
+            # False and use load_vars, load_duals, etc.
             if have_soln:
                 soln_variables = soln.variable
                 soln_constraints = soln.constraint
 
+                if extract_duals or extract_slacks:
+                    xpress_cons = list(self._solver_con_to_pyomo_con_map.keys())
+                    for con in xpress_cons:
+                        soln_constraints[con.name] = {}
+
                 xpress_vars = list(self._solver_var_to_pyomo_var_map.keys())
-                var_vals = xprob.getSolution(xpress_vars)
+                try:
+                    var_vals = xprob.getSolution(xpress_vars)
+                    if extract_slacks:
+                        slacks = self._getSlacks(xprob, xpress_cons)
+                except xpress.ModelError:
+                    # Xpress 9.5.0 has new behavior for unbounded
+                    # problems that have mipsols > 0.  Previously
+                    # getSolution() would return a solution, but now
+                    # raises a ModelError (even though the deprecated
+                    # getmipsol() will return a solution).  We will try
+                    # to fall back on the [deprecated] getmipsol(), but
+                    # if it fails, we will raise the original exception.
```

Comment:
Is this worth reporting to the Xpress folks? It seems like undesirable behavior, to say the least, and they may eventually remove `getmipsol`.

Reply:
Yes - it is, once we get our tests passing again.  There were other changes that *looked* like compatibility issues until I got all the deprecation warnings silenced.  (I have some evidence that other deprecation paths are not actually 100% backwards compatible).

Reply:
Just so that I understand correctly: you are saying that the [MIPSOLS](https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/MIPSOLS.html) attribute returns a number greater than zero but still `getSolution()` raises an error?

Reply:
That is correct.  It's probably an unintentional regression [and admittedly a pretty weird edge case that we happened to have a test case for] -- filing a bug report upstream is on our backlog (we are still working / fighting infrastructure to try and get a release out this week)

Reply:
Thanks. We confirm that this is a defect in the Xpress C library. We will work on it.